precisely/iterable_matchers.py

```diff
@@ -139,4 +139,30 @@ def describe(self):
             ))
 
 
+def contains_only(matcher):
+    return IsSequenceContainsOnly(matcher)
+
+
+class IsSequenceContainsOnly(Matcher):
+
+    def __init__(self, matcher):
+        self._matcher = matcher
+
+    def match(self, actual):
+        values = list(actual)
+
+        if len(values) == 0:
+            return unmatched(_empty_iterable_description)
```

Comment:
By analogy with `all()`, I'd expect the empty iterable to always match.

Reply:
I've updated the code, but this does feel a little counter intuitive to me - for example

```
     assert_that([], all_elements(equal_to(42)))
```

Can we really say that every item in an empty list is equal to anything? ie If I want to check every item in a list is 42, I would be surprised if an empty list matched.